app/routes/models/searchResultsModel.ts

```diff
@@ -186,7 +206,7 @@ export const createModel = async (query: any, error?: any) => {
     counties: typeof query.counties === 'string' ? [query.counties]: query.counties
   }
 
-  const { matches, total } = await performSearch(searchString, requestedPage, filterBy, sortBy)
+  const { matches, total } = await performSearch(searchString, requestedPage, filterBy, sortBy, action)
```

Comment:
If you copy the code in the other handler then you can avoid some of this code to pass in action variable as it default to null in getPaymentData function.